### PR TITLE
AVX-16372: add support for approved_learned_cidrs

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 1]
```diff
@@ -1425,6 +1446,15 @@ func resourceAviatrixTransitGatewayRead(d *schema.ResourceData, meta interface{}
 	d.Set("enable_vpc_dns_server", goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AWSRelatedCloudTypes|goaviatrix.AzureArmRelatedCloudTypes|goaviatrix.AliCloudRelatedCloudTypes) && gw.EnableVpcDnsServer == "Enabled")
 	d.Set("enable_advertise_transit_cidr", gw.EnableAdvertiseTransitCidr)
 	d.Set("enable_learned_cidrs_approval", gw.EnableLearnedCidrsApproval)
+	if gw.EnableLearnedCidrsApproval {
+		transitAdvancedConfig, err := client.GetTransitGatewayAdvancedConfig(&goaviatrix.TransitVpc{GwName: gw.GwName})
+		if err != nil {
+			return fmt.Errorf("could not get advanced config for transit gateway: %v", err)
+		}
+		d.Set("approved_learned_cidrs", transitAdvancedConfig.ApprovedLearnedCidrs)
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_gateway.go]: TypeSet is no a primitive type. May do a check if this set is successfully.

File: docs/resources/aviatrix_transit_gateway.md [hunk 2]
```diff
@@ -298,6 +298,7 @@ The following arguments are supported:
 
 * `enable_learned_cidrs_approval` - (Optional) Switch to enable/disable encrypted transit approval for transit gateway. Valid values: true, false. Default value: false.
 * `learned_cidrs_approval_mode` - (Optional) Learned CIDRs approval mode. Either "gateway" (approval on a per gateway basis) or "connection" (approval on a per connection basis). Default value: "gateway". Available as of provider version R2.18+.
+* `approved_learned_cidrs` - (Optional) A list of approved learned CIDRs. Only valid when `enable_learned_cidrs_approval` is set to true. Example: ["10.250.0.0/16", "10.251.0.0/16"]. Available as of provider version R2.21+.
```
Comment (zhiqiangjin) [inline docs/resources/aviatrix_transit_gateway.md]: A set of.

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 3]
```diff
@@ -280,6 +280,15 @@ func resourceAviatrixTransitGateway() *schema.Resource {
 					"connection, use the enable_learned_cidrs_approval attribute within the connection resource to " +
 					"toggle learned CIDR approval. Valid values: 'gateway' or 'connection'. Default value: 'gateway'.",
 			},
+			"approved_learned_cidrs": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsCIDR,
+				},
+				Optional:    true,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_gateway.go:289]: Is it computed also?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_transit_gateway.go (hunk 3 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_transit_gateway.go:289]: This is not computed.

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**